monitor/src/main/webapp/WEB-INF/views/admin/addApp.jsp

```diff
@@ -123,6 +132,71 @@
   </c:choose>
 </form:form>
 <script>
+$(document).ready(function() {
+	var currentTags = [];
+	
+	$("#tagString").autocomplete({
+		source : function(request, response) {
+			$.ajax({
+				url : "${pageContext.request.contextPath}/admin/apps/tags/getTagList",
+				dataType : "json",
+				data : {
+					term : request.term
+				},
+				success : function(data) {
+					response($.map(data, function(item){
+						return item.name;
+					}));
+				}
+			});
+		},
+		minLength : 1,
+		delay : 500,
+
+		//define select handler
+		select : function(event, ui) {
+			if (!currentTags.includes(ui.item.value)){
+				updateTagsField(ui.item.value);
+			}
+			$(this).val("");
+			// Prevent the default autocomplete behavior
+			return false;
+		},
+		focus: function(event, ui) {
+	        $(this).val(ui.item.label);
+	        return false;
+        }
+	}).keydown(function(event){
+		if (event.keyCode === $.ui.keyCode.ENTER) {
+			event.preventDefault();
+			var newTag = $(this).val();
+			if (!currentTags.includes(newTag)){
+				updateTagsField(newTag);
+			}
+			$(this).val("");
+		}
+	});
+	
+	function updateTagsField(tag) {
+		if (tag.trim() != ""){
+			var button = $('<button type="button" class="btn btn-default">' + tag + ' <span class="glyphicon glyphicon-remove"></span></button>');
+		    
+		    // Add an event listener to the button to remove the tag when clicked
+		    button.on('click', function() {
+		    	button.remove();
+			    var index = currentTags.indexOf(tag);
+			    if (index > -1) {
+			    	currentTags.splice(index, 1);
+			        $("#hidden-tags").val(currentTags.join(","));
```

Comment:
I believe you can just have multiple hidden fields with the same name (one for each tag), and that will be turned into an array or list when send to the controller.